src/Components/3DV/SceneView.stories.tsx

```diff
@@ -16,6 +35,231 @@ export default {
+    gizmoTransformItem: TransformInfo;
+    setGizmoTransformItem?: (gizmoTransformItem: TransformInfo) => void;
+}
+
+const FakePanel = (props: IFakePanelProps) => {
+    const { gizmoTransformItem, setGizmoTransformItem } = props;
+    const gizmoTransformItemDraftRef = useRef<TransformInfo>(null);
+
+    useEffect(() => {
+        if (gizmoTransformItem) {
+            gizmoTransformItemDraftRef.current = deepCopy(gizmoTransformItem);
+        }
+    }, [gizmoTransformItem]);
+
+    const onTransformChange = useCallback(
+        (event) => {
+            if (gizmoTransformItem) {
+                const { name, value } = event.target;
+                gizmoTransformItemDraftRef.current = deepCopy(
+                    gizmoTransformItem
+                );
+                const valueAsNumber = value;
+                switch (name) {
+                    case 'xPos':
+                        gizmoTransformItemDraftRef.current.position.x = valueAsNumber;
+                        break;
+                    case 'yPos':
+                        gizmoTransformItemDraftRef.current.position.y = valueAsNumber;
+                        break;
+                    case 'zPos':
+                        gizmoTransformItemDraftRef.current.position.z = valueAsNumber;
+                        break;
+                    case 'xRot':
+                        gizmoTransformItemDraftRef.current.rotation.x = valueAsNumber;
+                        break;
+                    case 'yRot':
+                        gizmoTransformItemDraftRef.current.rotation.y = valueAsNumber;
+                        break;
+                    case 'zRot':
+                        gizmoTransformItemDraftRef.current.rotation.z = valueAsNumber;
+                        break;
+                }
+                setGizmoTransformItem(gizmoTransformItemDraftRef.current);
+            }
+        },
+        [gizmoTransformItem]
+    );
+
+    const onTransformBlur = useCallback(() => {
+        setGizmoTransformItem({
+            position: {
+                x: Number(gizmoTransformItem.position.x),
+                y: Number(gizmoTransformItem.position.y),
+                z: Number(gizmoTransformItem.position.z)
+            },
+            rotation: {
+                x: Number(gizmoTransformItem.rotation.x),
+                y: Number(gizmoTransformItem.rotation.y),
+                z: Number(gizmoTransformItem.rotation.z)
+            }
+        });
+    }, [gizmoTransformItem]);
+
+    return (
+        <div style={{ padding: '20px' }}>
+            <Stack tokens={{ childrenGap: 12 }} horizontal>
+                <TextField
+                    label="X Position: "
+                    name="xPos"
+                    type="number"
+                    onChange={onTransformChange}
+                    onBlur={onTransformBlur}
+                    value={
+                        gizmoTransformItem
+                            ? '' + gizmoTransformItem.position.x
```

Comment:
hmm we should not need to cast it to string, the number values should be working in a text field value, no?

Reply:
![image](https://user-images.githubusercontent.com/55771758/179842933-8076f484-87ea-435d-9a70-9b97f379c90a.png)
I get a type error :( says number cannot be cast to string